sql-plugin/src/main/scala/org/apache/spark/sql/rapids/collectionOperations.scala

```diff
@@ -108,6 +109,110 @@ case class GpuMapConcat(children: Seq[Expression]) extends GpuComplexTypeMerging
     }
 }
 
+case class GpuListSlice(x: Expression, start: Expression, length: Expression)
```

Comment:
Is it better to call it `GpuSlice` since the operator name in Spark is `Slice`? I think in jni we can keep the name consistent with cudf, but in spark-rapids we can keep the name consistent with Spark at the same time.